exchange.c

```diff
@@ -327,6 +328,7 @@ domain_build_plan(ptrdiff_t nlimit, int (*layoutfunc)(int p), ExchangePlan * pla
     {
         if(package >= nlimit) {insuf = 1; break; }
         if(!P[n].OnAnotherDomain) continue;
+        if(P[n].IsGarbage) continue;
```

Comment:
If we have this line then the line above can be an assertion?

Reply:
My understanding is that build_plan counts particles and domain_exchange copies them into the buffer: so you need to skip garbage in both places or the counts will not correspond to the buffer sizes. Is that not right?

Reply:
But WillExport of garbage is always False.

Reply:
Ah, indeed! Absolutely - this line is an assertion. I will fix that.